src/ResultsHelpers.ts

```diff
@@ -0,0 +1,839 @@
+  } else if (result && typeof result._type === 'string' && result._type.includes('QDM::')) {
+    // If there isn't a description, use the type name as a fallback.  This mirrors the frontend where we do
+    // result.constructor.name.
+    const description = result.description ? `${result.description}\n` : `${result._type.replace('QDM::', '')}\n`;
+    let startDateTime = null;
+    let endDateTime = null;
+    let startTimeString = '';
+    let endTimeString = '';
+    // Start time of data element is start of relevant period, if data element does not have relevant period, use authorDatetime
+    if (result.relevantPeriod) {
+      if (result.relevantPeriod.low) {
+        startDateTime = result.relevantPeriod.low;
+      }
+      if (result.relevantPeriod.high) {
+        endDateTime = result.relevantPeriod.high;
+      }
+    } else if (result.prevalencePeriod) {
+      if (result.prevalencePeriod.low) {
+        startDateTime = result.prevalencePeriod.low;
+      }
+      if (result.prevalencePeriod.high) {
+        endDateTime = result.prevalencePeriod.high;
+      }
+    } else if (result.authorDatetime) {
+      // TODO: start result string will need to be updated to AUTHORED once bonnie frontend
+      // updates its pretty printer to do so.
+      startDateTime = result.authorDatetime;
+    }
+
+    if (startDateTime) {
+      startTimeString = `START: ${moment.utc(startDateTime.toString()).format('MM/DD/YYYY h:mm A')}\n`;
+    }
+    // If endTime is the infinity dateTime, clear it out because we do not want to export it
+    if (endDateTime && endDateTime.year !== 9999) {
+      endTimeString = `STOP: ${moment.utc(endDateTime.toString()).format('MM/DD/YYYY h:mm A')}\n`;
+    }
+    const system = result.dataElementCodes[0].system;
+    // TODO: Sort out a better way to have a friendly system display for FHIR codes
+    const codeDisplay =
+      result.dataElementCodes && result.dataElementCodes[0] ? `CODE: ${system} ${result.dataElementCodes[0].code}` : '';
+    // Add indentation
+    const returnString = `${description}${startTimeString}${endTimeString}${codeDisplay}`;
+    return returnString.replace(/\n/g, `\n${currentIndentation}${keyIndentation}`);
+  } else if (result instanceof String || typeof result === 'string') {
+    return `"${result}"`;
+  } else if (result instanceof Array) {
+    prettyResultReturn = result.map((value: any) => prettyResult(value, indentLevel, keyIndent));
+    return `[${prettyResultReturn.join(`,\n${currentIndentation}${keyIndentation}`)}]`;
+  } else if (result instanceof Object) {
+    // if the object has it's own custom toString method, use that instead
+    if (typeof result.toString === 'function' && result.toString !== Object.prototype.toString) {
+      return result.toString();
+    }
+    prettyResultReturn = '{\n';
+    const baseIndentation = Array(3).join(' ');
+    const sortedKeys = Object.keys(result)
+      .sort()
+      .filter(key => key !== '_type' && key !== 'qdmVersion');
```

Comment:
Will qdmVersion ever be a thing?

Reply:
Nope. This code is to print info about a result that contains some element of data originally from QDM. We wanted to filter out the internal/not relevant pieces. We may have to do similar work with the FHIR data models we use. This function isn't important for the initial release of Measure Highlighting. Should I add a TODO to address this later for FHIR types?

Reply:
Yeah that makes sense to me. TODO for FHIR types is good

Reply:
TODO added in 3878c44.